src/Lime.Transport.SignalR/ClientSignalRTransport.cs

```diff
@@ -0,0 +1,179 @@
+        public ClientSignalRTransport(
+            IEnvelopeSerializer envelopeSerializer,
+            ITraceWriter traceWriter = null) : base(Channel.CreateUnbounded<string>(),
+                                                    envelopeSerializer,
+                                                    traceWriter)
+        {
+        }
+
+        /// <summary>
+        /// Gets the remote endpoint address when the underlying transport is WebSockets; Otherwise, returns an empty string.
+        /// </summary>
+        public override string RemoteEndPoint
+        {
+            get
+            {
+                try
+                {
+                    return _hubConnection.GetRemoteEndpoint()?.ToString() ?? string.Empty;
+                }
+                catch (MissingMemberException ex)
+                {
+                    TraceWriter.TraceIfEnabledAsync(ex.ToString(), DataOperation.Error);
+                    return string.Empty;
+                }
+            }
+        }
+
+        /// <summary>
+        /// Gets the local endpoint address when the underlying transport is WebSockets; Otherwise, returns an empty string.
+        /// </summary>
+        public override string LocalEndPoint
+        {
+            get
+            {
+                try
+                {
+                    return _hubConnection.GetLocalEndpoint()?.ToString() ?? string.Empty;
+                }
+                catch (MissingMemberException ex)
+                {
+                    TraceWriter.TraceIfEnabledAsync(ex.ToString(), DataOperation.Error);
+                    return string.Empty;
+                }
+            }
+        }
+
+        /// <inheritdoc />
+        public override IReadOnlyDictionary<string, object> Options => new Dictionary<string, object>
+        {
+            [nameof(_hubConnection.HandshakeTimeout)] = _hubConnection.HandshakeTimeout,
+            [nameof(_hubConnection.ServerTimeout)] = _hubConnection.ServerTimeout,
+            [nameof(_hubConnection.KeepAliveInterval)] = _hubConnection.KeepAliveInterval
+        };
+
+        /// <inheritdoc />
+        /// <exception cref="ObjectDisposedException">This instance has already been disposed.</exception>
+        public override async Task<Envelope> ReceiveAsync(CancellationToken cancellationToken)
+        {
+            ThrowIfNotConnected();
+
+            return await base.ReceiveAsync(cancellationToken).ConfigureAwait(false);
+        }
+
+        /// <inheritdoc />
+        /// <exception cref="ObjectDisposedException">This instance has already been disposed.</exception>
+        public override async Task SendAsync(Envelope envelope, CancellationToken cancellationToken)
+        {
+            ThrowIfNotConnected();
+
+            await base.SendAsync(envelope, cancellationToken).ConfigureAwait(false);
+
+            string envelopeSerialized = EnvelopeSerializer.Serialize(envelope);
```

Comment:
Use `serializedEnvelope` to keep consistency with other transports.
```suggestion
            string serializedEnvelope = EnvelopeSerializer.Serialize(envelope);
```